src/xdp-app-info-host.c

```diff
@@ -200,6 +202,7 @@ xdp_app_info_host_new_full (const char *app_id,
                            /* supports_opath */ TRUE,
                            /* has_network */ TRUE,
                            /* requires_pid_mapping */ FALSE);
+  app_info_host->registered = registered;
```

Comment:
The XdpAppInfoTest also can be registered which is needed for the tests and they now have divergent behavior.

Reply:
I've reworked XdpAppInfo internals quite intensely in this MR, and I think that's about as far as I can go without turning this PR into a monstrosity. Should make it easier to do any follow-up if necessary.